### PR TITLE
Page GitHub repo discovery via the Link header (>1000 repos)

### PR DESCRIPTION
**Pull request by kamilon**
## What

Makes GitHub repository discovery (used by both **manual import** and the **background auto-add** pass) page through **all** of an account's repos instead of stopping at a hard 10-page (1000-repo) cap. It now follows GitHub's `Link` response header (`rel="next"`) with a 50-page (~5000-repo) safety cap.

## How

- **`lib/repo_discovery_service.dart`** — `_fetchGithubPaged` starts at `?per_page=100` and loops on the `rel="next"` URL from the `Link` header (parsed by the new `_nextLink`) rather than incrementing a page counter. `truncated` is set only when the safety cap (or the cycle guard) trips while more pages remain.
  - **Security:** `_nextLink` only returns **same-origin `https://api.github.com`** URLs — a next link to any other host/scheme is rejected — so a hostile or proxying response can't redirect the token-bearing request (and its `Authorization` header) to another host.
  - **Cycle guard:** visited next-URLs are tracked, so a self-referential `next` stops immediately (as truncated) rather than looping to the cap.
- **`lib/repo_discovery_page.dart`** — the import UI's truncation notice no longer hard-codes "first 1000 repositories".
- **ADO** listing is unchanged — its `/git/repositories` endpoint returns all repos in one response (no pagination).

## Review gate

Pre-open review by code-review + rubber-duck. Rubber-duck flagged a **cross-host token-leak/SSRF** risk in following `Link` URLs unconditionally — addressed by the same-origin `https://api.github.com` validation above (with a regression test).

## Tests

`test/repo_discovery_test.dart`: multi-page Link following (100 + 5 → 105 repos, not truncated), single-page stop (exactly one request when no Link header), and **cross-host next-link rejection** (only `api.github.com` is contacted). Two `auto_add_service_test` mocks updated for the new initial request (no `page=1`).

`dart format` clean, `flutter analyze --fatal-infos` clean, all 332 tests pass.

### CODE REVIEW

File: lib/repo_discovery_service.dart [hunk 1]
```diff
@@ -190,13 +206,40 @@ class RepoDiscoveryService {
       final body = jsonDecode(response.body);
       if (body is! List || body.isEmpty) break;
       all.addAll(parseGithubRepos(body));
-      if (body.length < 100) break;
-      // A full final page means there may be more we did not fetch.
-      if (page == 10) truncated = true;
+      // Follow the server's rel="next" link; its absence means the last page.
+      next = _nextLink(response.headers['link']);
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/repo_discovery_service.dart]: Fixed in d69c31e — `_nextLink` now returns `(url, rejected)`; when a `rel="next"` is present but rejected (cross-host/unparseable), `rejected` is true and the loop sets `truncated = true`, so the UI warns that repos were omitted instead of silently stopping. The cross-host test now also asserts `truncated`.

File: test/repo_discovery_test.dart [hunk 2]
```diff
@@ -91,4 +91,112 @@ void main() {
...
+          {
+            'name': '$prefix$i',
+            'owner': {'login': 'acme'},
+          },
+      ];
+      final client = MockClient((request) async {
+        if (request.url.path != '/user/repos') {
+          return http.Response('[]', 200);
+        }
+        final page = request.url.queryParameters['page'];
```
[the remark this answers is not in the document]
Comment (kamilon) [reply test/repo_discovery_test.dart]: Fixed in 19096eb — the test now returns `fail('unexpected request: …')` for anything other than the two expected /user/repos pages, so a wrong-endpoint or extra call fails the test instead of being masked by an empty 200.

File: lib/repo_discovery_page.dart [hunk 3]
```diff
@@ -366,8 +366,9 @@ class _RepoDiscoveryPageState extends State<RepoDiscoveryPage> {
           if (_truncated) ...[
             const SizedBox(height: 12),
             Text(
-              'Showing the first 1000 repositories. Narrow the results by '
-              'organization to see the rest.',
+              'Some repositories were omitted because the account has a very '
+              'large number of them. Narrow the results by organization to see '
+              'the rest.',
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/repo_discovery_page.dart]: Fixed in 19096eb — reworded to 'Some repositories were omitted (the account has too many to list, or its pagination could not be followed)…', which covers the cap, cross-host/unparseable link, and cycle cases while keeping the same 'narrow by organization' guidance.
